sdk/eventhub/Azure.Messaging.EventHubs/src/Core/Guard.cs

```diff
@@ -151,7 +170,28 @@ internal static class Guard
         ///
         /// <param name="argumentName">The name of the argument being considered.</param>
         /// <param name="argumentValue">The value of the argument to verify.</param>
-        /// <param name="minimumValue">The maximum to use for comparison; <paramref name="argumentValue"/> must be greater than or equal to this value.</param>
+        /// <param name="minimumValue">The minimum to use for comparison; <paramref name="argumentValue"/> must be greater than or equal to this value.</param>
+        /// <param name="maximumValue">The maximum to use for comparison; <paramref name="argumentValue"/> must be less than or equal to this value.</param>
+        ///
+        public static void ArgumentInRange(string argumentName,
```

Comment:
Given the small amount of code, it was clearer and less work to use an override rather than make it generic and apply the correct constraints for numeric.